### ISSUE TITLE
NumberControls should have a "snappy" slider 

### ISSUE THREAD

**Issue by zepumph** [phetsims/gravity-force-lab#173]
In a PhET-iO design meeting for GFL, @kathy-phet mentioned that the number control slider should snap to 10, and the tweakers should be able to fine tune it to the ones place.

@kathy-phet mentioned this is what we are doing in more recent sim designs. @arouinfar and @terracoda what do you think?

**Comment by zepumph** [comment]
From https://github.com/phetsims/gravity-force-lab/issues/76

**Comment by arouinfar** [comment]
@zepumph I like this suggestion. This is indeed a common design pattern we've used across many sims, particularly when the slider range is large. I believe the relevant option is called `constrainValue`.

**Comment by zepumph** [comment]
Does this mean that you will not be able to get at the smaller values? Or that you will snap with the slider, and still be able to access the single digits with the tweakers?

**Comment by kathy-phet** [comment]
I think we agreed today that we would have a minimum of 0, so no single digits. It also makes the range of the force much better behaved. And we didn’t see a pedagogical reason that we needed to go to a value of 1.
K

Sent from my iPhone

On Aug 22, 2019, at 7:16 PM, Michael Kauzmann <notifications@github.com<mailto:notifications@github.com>> wrote:


Does this mean that you will not be able to get at the smaller values? Or that you will snap with the slider, and still be able to access the single digits with the tweakers?

**Comment by terracoda** [comment]
I feel this issue is related to #133, and I do indeed agree the sliders should be snappy.

Starting at zero is fine with me if that makes for the sim. I would, however, need to double check the described values. As of right now there is no described value of zero.

**Comment by zepumph** [comment]
>I think we agreed today that we would have a minimum of 0, so no single digits.

I thought we agreed to have a minimum of 10.

**Comment by zepumph** [comment]
The GFL sliders now snap to 10. The current minimum is 10, and the keyboard shift key delta is 10. @arouinfar @terracoda please review.

**Comment by kathy-phet** [comment]
@zepumph - The tweekers should be set to steps of "1". There is a way to set the snappy different from the tweekers.

**Comment by zepumph** [comment]
@kathy-phet I implemented that above. The issue is that currently there is no way to make that possible for accessibility. I created https://github.com/phetsims/scenery-phet/issues/528 to fix that. But for now this issue should block a11y publication of this sim (not that was coming up anytime soon).

**Comment by zepumph** [comment]
@kathy-phet please review. Is this good enough to give to a client today?

**Comment by kathy-phet** [comment]
I think this looks nice. Good enough for the dev delivery. Marking for review by @arouinfar and @ariel-phet when they are around. Please see what you think of the arrow scaling in GFL and GFLB now.

**Comment by zepumph** [comment]
Review of the arrow scaling can happen over in https://github.com/phetsims/inverse-square-law-common/issues/76. 

It seems like the snappy slider is looking good for gravity-force-lab, and there have been no complaints from our clients. Closing this issue.
